python/cuxfilter/tests/charts/core/non_aggregate/test_utils.py

```diff
@@ -0,0 +1,122 @@
+import pytest
+import cudf
+import numpy as np
+from cudf.testing import assert_series_equal
+
+# Assume the function is in this path relative to the tests
+from cuxfilter.charts.core.non_aggregate.utils import point_in_polygon
+
+
+@pytest.fixture
+def sample_points_df():
+    """Create a sample cudf DataFrame for testing."""
+    data = {
+        "x": [
+            0.5,
+            1.5,
+            0.5,
+            1.5,
+            0.0,
+            2.0,
+            1.0,
+            1.0,
+        ],
+        # Inside, Inside, Outside, Outside, Boundary, Boundary,
```

Comment:
Be very cautious with boundary testing, as numerical precision issues can cause the behavior to differ in unstable ways, causing tests to be flaky as versions/platforms change. In my understanding, a user will be doodling a polygon on a map and the boundary behavior won't matter so precisely. I would advise against doing any boundary testing here.

Reply:
that's a fair point, I agree with this